src/color.rs

```diff
@@ -48,6 +48,16 @@ impl ColorType {
         }
     }
 
+    /// Returns if there is an alpha channel.
+    pub fn has_alpha(self) -> bool {
+        use ColorType::*;
+        match self {
+            L8 | L16 | Rgb8 | Bgr8 | Rgb16 => false,
+            La8 | Rgba8 | Bgra8 | La16 | Rgba16 => true,
+            __NonExhaustive(marker) => match marker._private {},
```

Comment:
What dose `__NonExhaustive(marker) => match marker._private {},` do?

Reply:
The `__NonExhaustive` variant emulates the `#[non_exhaustive]` attribute, which we can't use due to the minimum supported compiler version policy. It's defined [here in utils](https://github.com/image-rs/image/blob/95202a4af089ed22c17348992d2678dab77451c0/src/utils/mod.rs#L78-L95) which has an explanation for how this works and what alternatives were considered.

Reply:
The explicit match here is used to prove to the compiler that the variant can in fact never be constructed, without having to rely on `unreachable` (which may compile to a panic) or the unsafe `unreachable_unchecked`.

Reply:
Nice, Thanks for the explanation.